src/stream/src/executor/backfill/upstream_table/snapshot.rs

```diff
@@ -92,52 +88,43 @@ impl<T> UpstreamTableReader<T> {
 }
 
 impl UpstreamTableRead for UpstreamTableReader<ExternalStorageTable> {
-    type BinlogOffsetFuture<'a> =
-        impl Future<Output = StreamExecutorResult<Option<CdcOffset>>> + 'a;
-    type SnapshotStream<'a> = impl Stream<Item = StreamExecutorResult<Option<StreamChunk>>> + 'a;
-
-    fn snapshot_read(&self, args: SnapshotReadArgs) -> Self::SnapshotStream<'_> {
-        #[try_stream]
-        async move {
-            let primary_keys = self
-                .inner
-                .pk_indices()
-                .iter()
-                .map(|idx| {
-                    let f = &self.inner.schema().fields[*idx];
-                    f.name.clone()
-                })
-                .collect_vec();
-
-            tracing::debug!(
-                "snapshot_read primary keys: {:?}, current_pos: {:?}",
-                primary_keys,
-                args.current_pos
-            );
-
-            let row_stream = self.inner.table_reader().snapshot_read(
-                self.inner.schema_table_name(),
-                args.current_pos,
-                primary_keys,
-            );
-
-            pin_mut!(row_stream);
-
-            let mut builder =
-                DataChunkBuilder::new(self.inner.schema().data_types(), args.chunk_size);
-            let chunk_stream = iter_chunks(row_stream, &mut builder);
-            #[for_await]
-            for chunk in chunk_stream {
-                yield chunk?;
-            }
+    #[try_stream(ok = Option<StreamChunk>, error = StreamExecutorError)]
+    async fn snapshot_read(&self, args: SnapshotReadArgs) {
```

Comment:
Looks like we can also use `#[try_stream]` in `impl Executor` to avoid the indirect `execute_inner`. 🤔